gaybot.py

```diff
@@ -114,25 +107,22 @@ def mystery_box_event(self):
                 mystery_box.cleanup()
 
 
-
     def lottery_event(self):
         print ('[lottery] is being drawn')
         self.message('The winning lottery ticket will be drawn in 1 minute! Buy your tickets for 5 points with !buytickets <qty>.')
         sleep(6)
         winner = lottery.draw()
         self.message('The winning lottery ticket has been drawn and...')
         sleep(3)
-        if winner == '':
-            self.message('Nobody won the lottery PepeREE')
-            print ('[lottery] no winners')
-        else:
-            self.message(winner + ' won the lottery! You won ' + format(lottery.get_value(), ',d') + ' points, giving you ' + 
+        if winner:

```

Comment:
i'm not entirely sure why this makes sense. winner is a string and not a boolean, so what is this if statement checking for?

Reply:
In Python, data types can be checked in **if** statements directly.
You can try this out in a shell and see the results.

Strings return _False_ if it is empty (including ""), _True_ if it has a value.
Integers return _False_ if they are zero, _True_ otherwise.
Lists return _False_ if they are empty, _True_ otherwise.
etc.

So this if statement checks if winner has a value assigned to it. If it does, then a winner is found. You'll notice that in the draw() method, I also removed the _return ""_. That's because if it doesn't return a value, there is no winner, so _if winner_ will return _False_. I think.